src/reverse/RTTIHelper.cpp

```diff
@@ -770,6 +771,53 @@ sol::variadic_results RTTIHelper::ExecuteFunction(RED4ext::CBaseFunction* apFunc
     return results;
 }
 
+bool RTTIHelper::ExecuteFunction(RED4ext::CBaseFunction* apFunc, RED4ext::IScriptable* apContext,
+                                 TiltedPhoques::Vector<RED4ext::CStackType>& aArgs, RED4ext::CStackType& aResult) const
+{
+    constexpr auto NopOp = 0;
+    constexpr auto ParamOp = 27;
+    constexpr auto ParamEndOp = 38;
+    constexpr auto MaxCodeSize = 264;
+    constexpr auto PointerSize = sizeof(void*);
+
+    char code[MaxCodeSize];
+    RED4ext::CStackFrame frame(nullptr, code);
+
+    for (uint32_t i = 0; i < apFunc->params.size; ++i)
+    {
+        const auto& param = apFunc->params[i];
+        const auto& arg = aArgs[i];
+
```

Comment:
I would add a check for current `frame.code` pointer, if someone supplies too many args it will overflow the buffer.

Also, do you think it is a good idea to port this to the SDK?

Reply:
Yeah, I think it makes sense to add to SDK. And if we want to make the function calling safer, then we have to add checks to all methods, since the engine doesn't check anything. This is already safer, because it relies on the actual params from RTTI, not on passed args. Whereas the original `ExecuteNative` just writes all the args you pass with `CStack` without even checking what the function accepts and without checking the buffer length.